ocaml/xapi/xapi_xenops.ml

```diff
@@ -491,20 +484,33 @@ module MD = struct
         (of_pvs_proxy ~__context vif)
         (Pvs_proxy_control.find_proxy_for_vif ~__context ~vif:vif_ref)
     in
+    let vlan = match pifs with
+      | [] -> None
+      | pif :: _ ->
+        let vlan = Db.PIF.get_VLAN ~__context ~self:pif in
+        if vlan < 0L then None else Some vlan
+    in
+    let backend =
+      match Xapi_vif_helpers.get_backend ~__context ~self:vif_ref with
+      | Network.Sriov {domain = 0; dev = 0; bus = 0; fn = 0} ->
```

Comment:
I don't really like the 0s as a special case here. We should just raise the error in `get_backend`.

Reply:
Fixed.